Add json_build_object

yirenlu92 opened the pull request:
Add postgres documentation for `json_build_object` function
https://neon-next-git-fork-yirenlu92-add-function-neondatabase.vercel.app/docs/functions/json_build_object

ruf-io commented:
This page currently ends rather abruptly. 

Was there anything that came up in the creation of the examples that would be useful to highlight in an "additional considerations" catch-all section at the end?

yirenlu92 commented:
> This page currently ends rather abruptly.
> 
> Was there anything that came up in the creation of the examples that would be useful to highlight in an "additional considerations" catch-all section at the end?

We can add a "additional considerations" section including some points about performance, etc.

bgrenon commented on content/docs/functions/json_build_object.md:
```diff
@@ -0,0 +1,265 @@
+---
+
+
+Use `json_build_object` to construct a JSON object from a set of key-value pairs, creating a JSON representation of a row or set of rows.
```
Maybe add some context or example, something like:
"This can increase performance compared to converting query results to JSON on the application side."

bgrenon commented on content/docs/functions/json_build_object.md:
````diff
@@ -0,0 +1,265 @@
+**users**
+
+```text
+| id |   name   | age |   city  
````
Content in rows 3 and 4 is duplicated. I'm not sure why.

yirenlu92 replied:
yeah this was a typo, deleted rows 3 and 4

bgrenon commented on content/docs/functions/json_build_object.md:
````diff
@@ -0,0 +1,265 @@
+
+
+{/*
+```sql
````
Why not just create one "products" table for this overall section and have it include the ratings? Instead of creating a second "products_with_ratings" table. That seems more straightforward, especially if someone wants to actually play with these queries.

bgrenon commented on content/docs/functions/json_build_object.md:
```diff
@@ -0,0 +1,265 @@
+           'rating', attributes->>'rating'
+       )
+   ) AS details
+FROM products_with_rating
```
Small thing, but I did find this confusing at first to see this 2nd 'products_with_rating' table. See comment above.

bgrenon commented on content/docs/functions/json_build_object.md:
````diff
@@ -0,0 +1,265 @@
+|----------|-------------------------
+| Kitchen  | {"total_price" : 12.99}
+| Clothing | {"total_price" : 25.99}
+```
````
```suggestion
```
## Resources

- [PostgreSQL Documentation: JSON Functions and Operators](https://www.postgresql.org/docs/current/functions-json.html)

- [PostgreSQL Documentation: JSON Types](https://www.postgresql.org/docs/current/datatype-json.html)
```

bgrenon replied:
- We should include a Resources section on all pages as part of the template. 
- We might want to have these links go to our mirrored docs, but would like to see if PixelPoint can do that for us globally somewhere in the build or web server. This would let us turn it off globally later if we move away from mirrored docs.
- Any other relevant links, it would be good to add also.

bgrenon reviewed (commented):
Four general comments:

- I think it's a good idea to lightly touch on the "why" of the function, some small description of a practical application or example, even in these reference function docs. I added a suggestion inline.
- Do we want to consider adding a styled Neon-specific section here to drive traffic and sign-ups? We could create a re-usable insert and include it in the template. Something about signing up and creating a project to try out this function. We could ask PixelPoint to style it in a way that does not detract from the pure postgres reference aspect of the doc.
- What about exposing the CREATE and INSERT statements in the published page? Some similar docs do start off with a "create this table to try things out" section. It might add too much noise, but raising it here for discussion. Especially if we decide to add a "try this in Neon" section, including the samples would make it easier to test drive the function in the Neon console.
- We should add a Resources section with links to the official docs as part of the template. I included an example inline.

yirenlu92 commented:
> Four general comments:
> 
> * I think it's a good idea to lightly touch on the "why" of the function, some small description of a practical application or example, even in these reference function docs. I added a suggestion inline.

ok, done.

> * Do we want to consider adding a styled Neon-specific section here to drive traffic and sign-ups? We could create a re-usable insert and include it in the template. Something about signing up and creating a project to try out this function. We could ask PixelPoint to style it in a way that does not detract from the pure postgres reference aspect of the doc.

Happy to add this, but need you to provide the insert - is it interactive?

> * What about exposing the CREATE and INSERT statements in the published page? Some similar docs do start off with a "create this table to try things out" section. It might add too much noise, but raising it here for discussion. Especially if we decide to add a "try this in Neon" section, including the samples would make it easier to test drive the function in the Neon console.

Fine with us, let us know how you guys decide on this.

> * We should add a Resources section with links to the official docs as part of the template. I included an example inline.

Can you take a look at the section we have added and let us know if that is appropriate?

bgrenon commented:
@yirenlu92 @danieltprice I'll get started

> > Four general comments:
> > 
> > * I think it's a good idea to lightly touch on the "why" of the function, some small description of a practical application or example, even in these reference function docs. I added a suggestion inline.
> 
> ok, done.
> 
> > * Do we want to consider adding a styled Neon-specific section here to drive traffic and sign-ups? We could create a re-usable insert and include it in the template. Something about signing up and creating a project to try out this function. We could ask PixelPoint to style it in a way that does not detract from the pure postgres reference aspect of the doc.
> 
> Happy to add this, but need you to provide the insert - is it interactive?
> 
> > * What about exposing the CREATE and INSERT statements in the published page? Some similar docs do start off with a "create this table to try things out" section. It might add too much noise, but raising it here for discussion. Especially if we decide to add a "try this in Neon" section, including the samples would make it easier to test drive the function in the Neon console.
> 
> Fine with us, let us know how you guys decide on this.
> 
> > * We should add a Resources section with links to the official docs as part of the template. I included an example inline.
> 
> Can you take a look at the section we have added and let us know if that is appropriate?

The resources section looks good.

@yirenlu92 @danieltprice I'll get started on an insert we can use to point user to Neon. Here is a link to a doc that includes the CREATE and INSERT to setup the data before the samples start. Something like this seems like a good approach:
https://rudra.dev/posts/generate-beautiful-json-from-postgresql/

bgrenon reviewed (approved):
Looks great!
